test/prif_image_queries_test.F90

```diff
@@ -1,61 +1,86 @@
-        result_ = assert_that(image_status == 0 .or. &
-                              image_status == PRIF_STAT_FAILED_IMAGE .or. &
-                              image_status == PRIF_STAT_STOPPED_IMAGE, "permitted image status")
-    end function
-
-    function valid_image_list(nums) result(result_)
-        integer, allocatable, intent(in) :: nums(:)
-        type(result_t) :: result_
-        integer i, ni
-
-        call prif_num_images(num_images=ni)
-        result_ = assert_that( allocated(nums) .and. size(nums) <= ni .and. &
-                               all([(nums(i) >= 1 .and. nums(i) <= ni, i = 1, size(nums))]) .and. &
-                               all([(nums(i) < nums(i+1), i = 1, size(nums)-1)]), &
-                               "valid stopped images")
-    end function
-
-    function check_stopped_images() result(result_)
-        type(result_t) :: result_
-        integer, allocatable :: nums(:)
-
-        call prif_stopped_images(stopped_images=nums)
-        result_ = valid_image_list(nums)
-    end function
-
-    function check_failed_images() result(result_)
-        type(result_t) :: result_
-        integer, allocatable :: nums(:)
-
-        call prif_failed_images(failed_images=nums)
-        result_ = valid_image_list(nums)
-    end function
-
-end module caf_image_queries_test
+
+  pure function subject() result(test_subject)
+    character(len=:), allocatable :: test_subject
+    test_subject = "PRIF image query procedures"
+  end function
+
+  function results() result(test_results)
+    type(test_result_t), allocatable :: test_results(:)
+    type(prif_image_queries_test_t) prif_image_queries_test
+
+    test_results = prif_image_queries_test%run([ &
+       test_description_t("providing valid prif_image_status()", usher(check_image_status)) &
+      ,test_description_t("providing valid prif_stopped_images()", usher(check_stopped_images)) &
+      ,test_description_t("providing valid prif_failed_images()", usher(check_failed_images)) &
+    ])
+  end function
+
+  function check_image_status() result(diag)
+      type(test_diagnosis_t) :: diag
+      integer(c_int) :: image_status
+      
+      call prif_image_status(1, image_status=image_status)
+      diag = .expect. (any(image_status == [0, PRIF_STAT_FAILED_IMAGE, PRIF_STAT_STOPPED_IMAGE])) & ! TODO: replace with .any. once Juliennes supports it
+                        // "permitted image status"
+  end function
+
+  function valid_image_list(nums) result(diag)
+      integer, allocatable, intent(in) :: nums(:)
+      type(test_diagnosis_t) :: diag
+      integer ni
+
+      call prif_num_images(num_images=ni)
+      diag = &
+         .expect. allocated(nums) &
+         .also. (size(nums) .isAtMost. ni) &
+         .also. (.all. (nums .isAtLeast. 1)) &
+         .also. (.all. (nums .isAtMost. ni)) &
+         .also. (.all. (nums(1:size(nums)-1) .lessThan. nums(2:size(nums)))) // "valid stopped image"
```

Comment:
Analogously to the incremental diagnosis idiom style described above, I have a style preference here to place the relevant test expression (the "signal") at the beginning of each line, and the diagnostic aggregation operators (the "noise") at the end of each line. Ex:

```suggestion
      diag = &
         .expect. allocated(nums) .also. &
         (size(nums) .isAtMost. ni) .also. &
         (.all. (nums .isAtLeast. 1)) .also. &
         (.all. (nums .isAtMost. ni)) .also. &
         (.all. (nums(1:size(nums)-1) .lessThan. nums(2:size(nums)))) // "valid stopped image"
```

I acknowledge it's a subjective cosmetic distinction, but I think it slightly improves readability of the test's domain semantics.